### DISCUSSION TITLE
Run Prisma migration on DB with existing tables

### DISCUSSION THREAD

**Discussion by ekafyi** [prisma/prisma#9631] [Q&A] [answered]
Is it possible to run Prisma migration on a Postgres DB with existing tables without affecting those tables?

**Detail:**

I run Metabase on Heroku + PostgreSQL DB using their [default starter](https://www.metabase.com/docs/latest/operations-guide/running-metabase-on-heroku.html). 

If I use Prisma to create new tables on that DB and run migrate, will it wipe/affect the Metabase-generated tables? Can i tell Prisma to simply ignore those? 

Assumptions:
- I will be responsible to make sure of no name collision/conflict (ie. I won't create/write to a table named like the Metabase tables)
- I will not be interacting with the Metabase tables from my Prisma app

**Comment by ryands17** [comment, marked as answer]
@ekafyi 👋 

I don't think that's possible. In this case you would be better off using [introspection](https://www.prisma.io/docs/concepts/components/introspection) and creating the tables yourself in the database.

**Reply by ekafyi** [reply]
Thanks @ryands17. I will try introspection so the Metabase tables are added to my Prisma schema.

**Reply by janpio** [reply]
You can _also_ start your project with introspecting the existing database tables, and then just not touch those in the Prisma schema. If we support all the features used in these tables, any future `migrate dev` should create Migrations that are only about the _new_ tables, and not the existing ones you did not change in the Prisma Schema.

Best way to test that: Create a Prisma project, configure your database connection, run `npx prisma db pull`, look it the schema matches the existing tables, then just run `npx prisma migrate dev --create-only` and observe it creating an empty migration file. If the file is not empty, you should come back an open an issue here so we can analyze what is going on :D

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
